src/pkcs11/framework-pkcs15.c

```diff
@@ -131,6 +131,8 @@ struct pkcs15_skey_object {
 	struct sc_pkcs15_skey *valueXXXX;
 };
 
+#define is_secret_key(obj) ((__p15_type(obj) & SC_PKCS15_TYPE_CLASS_MASK) == SC_PKCS15_TYPE_SKEY)
```

Comment:
use better names to differentiate from `is_skey`

Reply:
is_skey() macro is not used anywhere in the whole code base and it wouldn't even compile, because SC_PKCS15_TYPE_SKEY_OBJECT is not defined. Seems like this is why I added a new macro in the first place, didn't dare to change/remove existing code without understanding it. Both macros are not needed. Should we remove is_skey or fix it to compare to SC_PKCS15_TYPE_SKEY instead of SC_PKCS15_TYPE_SKEY_OBJECT, which naming would you prefer?

Reply:
I'd use `is_skey` and `is_skey_object`, i.e. I'd clearify the existing define.